src/sudo/cli/mod.rs

```diff
@@ -716,15 +701,20 @@ impl SudoOptions {
                         options.chdir = Some(SudoPath::from_cli_string(value));
                     }
                     "-E" | "--preserve-env" => {
-                        let split_value = || value.split(',').map(str::to_string);
                         match &mut options.preserve_env {
                             PreserveEnv::Nothing => {
-                                options.preserve_env = PreserveEnv::Only(split_value().collect())
+                                options.preserve_env = PreserveEnv::SelectedVars
                             }
                             PreserveEnv::Everything => {}
-                            PreserveEnv::Only(list) => list.extend(split_value()),
+                            PreserveEnv::SelectedVars => {}
```

Comment:
```suggestion
                            PreserveEnv::Everything | PreserveEnv::SelectedVars => {}
```
(github includes the line you already deleted in this diff for whatever reason)